atoma-service/src/middleware.rs

```diff
@@ -334,8 +335,24 @@ pub async fn verify_stack_permissions(
             StatusCode::UNAUTHORIZED
         })?;
     if available_stack.is_none() {
-        error!("No available stack with enough compute units");
-        return Err(StatusCode::UNAUTHORIZED);
+        let tx_digest_str = req_parts
+            .headers
+            .get("X-Tx-Digest")
+            .ok_or_else(|| {
+                error!("Tx digest header not found");
```

Comment:
Maybe change this to `Stack not found, tx digest header expected but not found` or something like that, because to someone unfamiliar with the code this error can sound like that tx digest should always be there.

Reply:
This makes a lot of sense, thank you !